### PR TITLE
fix(orchestrator): stop fetching workflow URI

### PR DESCRIPTION
**Pull request by caponetto**
https://issues.redhat.com/browse/FLPATH-1044

We were fetching each workflow URI directly from the workflow service when loading the workflow table.
If one of the services is not responsive, the loading operation fails. Not to mention that it adds one extra request for each item in the table. 

I've noticed that we were using the workflow URI only for authoring purposes so I'm removing this field in favor of the `format` field (`json|yaml`), which can be extracted from the workflow source, i.e., does not depend on fetching the workflow service directly.

I've also taken the opportunity to improve our API by simplifying some of the operations and removing others that are not being used anymore. There's still room for improvement and this PR is a proposal for a first step.

### CODE REVIEW

**Comment by caponetto** [comment]
@gciavarrini could you please help me review this PR? Especially the API part.

File: plugins/orchestrator-backend/src/service/router.ts [hunk 1]
```diff
@@ -171,41 +164,46 @@ function setupInternalRoutes(
...
-    await V1.getWorkflows(services.sonataFlowService, services.dataIndexService)
+  router.get('/workflows/:workflowId', async (req, res) => {
+    const {
+      params: { workflowId },
+    } = req;
+    await V1.getWorkflowById(services.sonataFlowService, workflowId)
       .then(result => res.status(200).json(result))
       .catch(error => {
-        res.status(500).send(error.message || 'internal Server Error');
+        res.status(500).send(error.message || 'Internal Server Error');
```
[the remark this answers is not in the document]
Comment (caponetto) [reply plugins/orchestrator-backend/src/service/router.ts:174]: It came from copy paste of other line. I can see there are some in uppercase and others in lowercase.
Let me create a const for this value `internal Server Error`.

File: plugins/orchestrator-common/src/openapi/openapi.yaml [hunk 2]
```diff
@@ -50,41 +50,48 @@ paths:
...
         '200':
           description: Success
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/WorkflowListResultDTO'
+                $ref: '#/components/schemas/WorkflowDTO'
         '500':
-          description: Error fetching workflow list
+          description: Error workflow by id
```
[the remark this answers is not in the document]
Comment (caponetto) [reply plugins/orchestrator-common/src/openapi/openapi.yaml:72]: Good catch, thanks!

File: plugins/orchestrator-backend/src/service/router.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (caponetto) [reply plugins/orchestrator-backend/src/service/router.ts:174]: I don't have a strong opinion about the case so I'm fine with the one you decide. But let's put it into a constant.
Also, let's merge your PR first, I can deal with the conflicts.

File: plugins/orchestrator-common/src/openapi/openapi.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (caponetto) [reply plugins/orchestrator-common/src/openapi/openapi.yaml]: Done in https://github.com/janus-idp/backstage-plugins/pull/1297/commits/56599c81f7e1f43b9bff97383abe6739f4906cf9

**Comment by caponetto** [comment]
@jkilzi PTAL

**Comment by jkilzi** [comment]
LGTM

**Review by batzionb — APPROVED** [review]

**Pull request merged**